connaisseur/validators/notaryv1/notaryv1_validator.py

```diff
@@ -118,9 +118,13 @@ async def __process_chain_of_trust(
 
         # load all trust data
         t_start = dt.datetime.now()
+
+        # even if a registry is public like e.g. docker.io, it might still
+        # require a token on repository level, specifying the scope (e.g. pull)
+        repo_token = await self.notary.get_auth(session, image)
```

Comment:
While this improves the runtime if there's auth, it adds another network call if the notary doesn't have auth as then `get_auth` retrieves the root data and discards it.

Could make this handling of root data special and have `get_root_trust_and_token` (better name welcome), which returns `(root_data, repo_token)`